File: JavaProto/app/src/main/java/com/example/javaproto/Obstacles.java
package com.example.javaproto;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;

import java.util.ArrayList;

public class Obstacles extends RectF{
    public float sizx;
    private boolean isOut=false;
    public int lane=0;
    public static float speed=15f;
    public Paint paint;
    public ArrayList<Integer> solution = new ArrayList<>();
    public Obstacles(float width, float size) {
        this.sizx=size;
        this.top=0;
        this.bottom=this.top+size;
        this.right=width-size/2;
        this.left=width+size/2;
    }

    public float getTop() {
        return top;
    }


    public void setPosition(float dim,int lane) {
        this.left = dim-sizx/2;
        this.right = dim+sizx/2;
        this.lane = lane;
    }

    public boolean isOut() {
        return isOut;
    }
    public RectF draw(Canvas canvas){
        if(canvas.getHeight()<top){
            isOut=true;
        }
        return this;
    }
    public void updatePosition(){
        this.top+=speed;
        this.bottom+=speed;
    }

}

File: MyApplication8/app/src/main/java/com/example/myapplication/Obstacles.kt
package com.example.myapplication

import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF

open class Obstacles(width: Float, size: Float): RectF() {
    companion object{
        var speed=15f
    }
    var sizx: Float = 0f
    private var isOut = false
    var lane: Int = 0
    var paint: Paint? = null
    var solution: ArrayList<Int> = ArrayList()
    init {
        this.sizx = size
        this.top=0f
        this.bottom = this.top + size
        this.right = width - size / 2
        this.left = width + size / 2
    }

    fun getTop(): Float {
        return top
    }
    fun setPosition(dim: Float, lane:Int) {
        this.left = dim - sizx / 2
        this.right = dim + sizx / 2
        this.lane=lane
    }
    fun isOut():Boolean{
        return isOut
    }
    fun draw(canvas: Canvas):RectF{
        if (canvas.height < top) {
            isOut = true
        }
        return this
    }
    fun updatePosition(){
        this.top+=speed
        this.bottom+=speed
    }
}